app/Filament/Resources/AchievementSetResource/RelationManagers/GameHashesRelationManager.php

```diff
@@ -0,0 +1,112 @@
+    }
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+
+            ]);
+    }
+
+    public function table(Table $table): Table
+    {
+        /** @var AchievementSet $achievementSet */
+        $achievementSet = $this->getOwnerRecord();
+
+        return $table
+            ->columns([
+                Tables\Columns\TextColumn::make('name')
+                    ->label('File Name'),
+
+                Tables\Columns\TextColumn::make('md5')
+                    ->label('MD5')
+                    ->fontFamily(FontFamily::Mono),
+            ])
+            ->filters([
+
+            ])
+            ->actions([
+                Tables\Actions\DetachAction::make(),
+            ])
+            ->headerActions([
+                Tables\Actions\Action::make('attachHashCompatibility')
+                    ->label('Mark Hash as Incompatible')
+                    ->color('danger')
+                    ->form([
+                        Forms\Components\Placeholder::make('warning')
+                            ->label('')
+                            ->content(fn () => new HtmlString("
+                                    🔴 PLEASE BE CAREFUL. Set types (bonus, specialty, exclusive) should automatically 
+                                    handle this for you. Only use this tool if you need to make a rare exception to the rules.
+                                    When you press Submit, the hash you've selected will no longer be eligible to receive 
+                                    this set's achievements when a player loads the hash in an emulator. This should almost
+                                    always be used VERY sparingly. If you find yourself frequently using this tool,
+                                    something is wrong. When in doubt, please ask for help!
+                                "
+                            )),
+
+                        Select::make('hash')
+                            ->label('Linked Hash')
+                            ->placeholder('Select hash')
+                            ->options(
+                                $achievementSet
+                                    ->games()
+                                    ->with('hashes')
+                                    ->get()
+                                    ->pluck('hashes')
+                                    ->flatten()
+                                    ->mapWithKeys(function (GameHash $hash) {
+                                        return [$hash->id => "{$hash->name} {$hash->md5}"];
+                                    })
+                                    ->toArray()
+                            ),
+                    ])
+                    ->action(function (array $data) use ($achievementSet) {
+                        $gameHashId = (int) $data['hash'];
+
+                        $achievementSet->incompatibleGameHashes()->attach($gameHashId);
```

Comment:
Error if hash is already marked as incompatible:
> SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1882-33972' for key 'set_hash_unique'

Reply:
Addressed in latest. Now after marking a hash as incompatible, it no longer appears as a select option. Additionally, I've added a few more guardrails in latest:

* The select field has been set to required.
* The action is disabled if there are no hashes to choose from.
* The "Detach" label and associated modal copy has been changed to be more clear.

Reply:
Cannot mark hash as incompatible if it's selected by default:
![image](https://github.com/user-attachments/assets/0750d7b0-8e3b-48e9-b729-2e686c7567bb)

Reply:
Fixed in latest.